iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2059,12 +2059,65 @@ TYPED_TEST(stringTyped_test, ConcatenateStringLiteralAndStringWorks)
 TEST(StringLiteralConcatenation, ConcatenateOnlyStringLiteralsWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "f591fb69-a8fa-4d8e-b42d-29573b4a20f7");
-    auto testString = iox::cxx::concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
+    auto testString = concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
     EXPECT_THAT(testString.capacity(), Eq(25U));
     EXPECT_THAT(testString.size(), Eq(25U));
     EXPECT_THAT(testString.c_str(), StrEq("FerdinandSpitzschnueffler"));
 }
 
+TYPED_TEST(stringTyped_test, ConcatenateEmptyStringAndNullCharReturnsStringWithSizeOne)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6a4374fb-b21f-4465-95dc-22b898079c81");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    auto result = concatenate(this->testSubject, '\0');
+
+    EXPECT_THAT(this->testSubject.capacity(), Eq(STRINGCAP));
```

Comment:
I think the capacity check is not needed. The capacity is a compile time value and cannot change. Of course, the `capacity` function could lie but I think we test this elsewhere.

Reply:
Checking the capacity, size and content of `this->testSubject` is the (non-exhaustive) attempt to check for unwanted side effects. The capacity is at the moment a compile time value but this could change when we implement [dynamic size types](https://github.com/eclipse-iceoryx/iceoryx/pull/912). Therefore, I would keep the capacity checks.